flight/Libraries/math/pid.c

```diff
@@ -190,7 +190,8 @@ float pid_apply_setpoint_antiwindup(struct pid *pid,
 		// If Ki is zero, do not change the integrator. We do not reset to zero
 		// because sometimes the accumulator term is set externally
 	} else {
-		pid->iAccumulator += err * (pid->i * dT);
+		float p = bound_sym(err * pid->p, 1.0f);
```

Comment:
This needs to be bounded and scaled based on the actual range, not assuming its -1..1 --- this would break nav.

Reply:
Are there any expectations regarding min_bound and max_bound, e.g. min_bound always zero or negative, and max_bound always zero or positive?

Reply:
Should we scale based on the larger boundary from zero?
```c
float s = MAX(fabsf(min_bound), fabsf(max_bound));
float p = bound_min_max(err * pid->p, min_bound, max_bound) / s;
```

Reply:
You could just let the caller pass in the absolute antiwindup bound.  0 means "don't do this step".

I'm not sure this logic is beneficial for all antiwindup use cases.  For example, the output of the antiwindup controllers is an attitude-- there's no comparable "slope" limitation.  Yes, it's good not to wind up the I term when you'd be exceeding the attitude limit.